src/kaleidoscope/hid.cpp

```diff
@@ -51,42 +52,172 @@
 namespace kaleidoscope {
 namespace hid {
 
-void initializeKeyboard() {
-  Keyboard.begin();
-  WITH_BOOTKEYBOARD {
-    BootKeyboard.begin();
-  }
+// Anonymous namespace for internal variables and helper functions
+namespace {
+
+
+// modifier_flag_mask is a bitmask of modifiers that we found attached to
+// keys that were newly pressed down during the most recent cycle with any new keypresses.
+
+// This is used to determine which modifier flags will be allowed to be added to the
+// current keyboard HID report. It gets set during any cycle where one or more new keys
+// is toggled on and presists until the next cycle with a newly detected keypress.
+
+static uint8_t modifier_flag_mask{0};
+
+
+
+// The functions in this namespace are primarily to solve the problem of rollover from a
+// key with a modifier flag (e.g. `LSHIFT(Key_T)`) to one without (e.g. `Key_H`), which
+// used to result in the mod flag being applied to keys other than the one with the
+// flag. By using `modifier_flag_mask`, we can mask out any modifier flags
+// that aren't attached to modifier keys or keys pressed in the most recent cycle,
```

Comment:
I would change "pressed" here to "pressed or held", since "pressed" is not well defined (often used to mean "toggled on").

Reply:
*nod* got it.